pkg/execution/state/redis_state/queue_processor.go

```diff
@@ -100,6 +100,11 @@ func (q *queue) Enqueue(ctx context.Context, item osqueue.Item, at time.Time) er
 		queueName = item.QueueName
 	}
 
+	// ensure both QueueName attributes are always set (ItemPartitions depends on this)
+	if queueName != nil && item.QueueName == nil {
+		item.QueueName = queueName
+	}
+
```

Comment:
Can we please return an error instead of enqueueing if the resulting queue name is a pointer to an empty string?

I'd like to detect unexpected inputs here, rather than having to discover them via debugging a problem deep in the queue.

Reply:
queueNames _can_ be nil as they're only used for overrides when we process system partitions, that's important. But they should never be an empty strings!